core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -113,6 +114,28 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
     return self();
   }
 
+  @Override
+  public ThisT toBranch(String branch) {
+    throw new UnsupportedOperationException(
+        String.format(
+            "Cannot commit to branch %s: %s does not support branch commits",
+            branch, this.getClass().getName()));
+  }
```

Comment:
This should be in the interface, not here.

Reply:
good point, fixed